R/pkg/R/mllib.R

```diff
@@ -632,3 +642,147 @@ setMethod("predict", signature(object = "AFTSurvivalRegressionModel"),
+#'    \item{seed}{integer seed for random number generation. Default: 0}
+#'    \item{numUserBlocks}{number of user blocks used to parallelize computation (> 0).
+#'                         Default: 10}
+#'    \item{numItemBlocks}{number of item blocks used to parallelize computation (> 0).
+#'                         Default: 10}
+#'    \item{checkpointInterval}{number of checkpoint intervals (>= 1) or disable checkpoint (-1).
+#'                              Default: 10}
+#'    }
+#'
+#' @param data A SparkDataFrame for training
+#' @param ratingCol column name for ratings
+#' @param userCol column name for user ids. Ids must be (or can be coerced into) integers
+#' @param itemCol column name for item ids. Ids must be (or can be coerced into) integers
+#' @param rank rank of the matrix factorization (> 0)
+#' @param reg regularization parameter (>= 0)
+#' @param maxIter maximum number of iterations (>= 0)
+
+#' @return \code{spark.als} returns a fitted ALS model
+#' @rdname spark.als
+#' @aliases spark.als,SparkDataFrame
+#' @name spark.als
+#' @export
+#' @examples
+#' \dontrun{
+#' df <- createDataFrame(ratings)
+#' model <- spark.als(df, "rating", "user", "item")
+#'
+#' # extract latent factors
+#' stats <- summary(model)
+#' userFactors <- stats$userFactors
+#' itemFactors <- stats$itemFactors
+#'
+#' # make predictions
+#' predicted <- predict(model, df)
+#' showDF(predicted)
+#'
+#' # save and load the model
+#' path <- "path/to/model"
+#' write.ml(model, path)
+#' savedModel <- read.ml(path)
+#' summary(savedModel)
+#'
+#' # set other arguments
+#' modelS <- spark.als(df, "rating", "user", "item", rank = 20,
+#'                     reg = 0.1, nonnegative = TRUE)
+#' statsS <- summary(modelS)
+#' }
+#' @note spark.als since 2.1.0
+setMethod("spark.als", signature(data = "SparkDataFrame"),
+          function(data, ratingCol = "rating", userCol = "user", itemCol = "item",
+                   rank = 10, reg = 1.0, maxIter = 10, ...) {
+
+            `%||%` <- function(a, b) if (!is.null(a)) a else b
+
+            args <- list(...)
+            numUserBlocks <- args$numUserBlocks %||% 10
+            numItemBlocks <- args$numItemBlocks %||% 10
+            implicitPrefs <- args$implicitPrefs %||% FALSE
+            alpha <- args$alpha %||% 1.0
+            nonnegative <- args$nonnegative %||% FALSE
+            checkpointInterval <- args$checkpointInterval %||% 10
+            seed <- args$seed %||% 0
+
+            features <- array(c(ratingCol, userCol, itemCol))
+            distParams <- array(as.integer(c(numUserBlocks, numItemBlocks,
+                                             checkpointInterval, seed)))
+
+            jobj <- callJStatic("org.apache.spark.ml.r.ALSWrapper",
+                                "fit", data@sdf, features, as.integer(rank),
+                                reg, as.integer(maxIter), implicitPrefs, alpha, nonnegative,
```

Comment:
here, it doesn't check if `rank` or `maxIter` is positive. Maybe some checks on R-side would be good.